### PR TITLE
add routes option by Divy (ready-to-merge)

### PR DESCRIPTION
**Pull request by littledivy**
```js
var Spring = require('./spring.js');
var { app } = new Spring({ 
name:"Hello", 
routes:[{
   url:"/",
   route:require('./routes/index')
  }]
})
```

### CODE REVIEW

**Comment by littledivy** [comment]
@dan-online Check if it works

**Comment by dan-online** [comment]
@DivySrivastava please add the test in the mocha file with adding a route and then testing it

**Comment by littledivy** [comment]
@dan-online I did that in the mocha file. Please check again

**Comment by littledivy** [comment]
@dan-online This is what I'm talking about
https://github.com/dan-online/spring.js/pull/1/commits/8f5d3f900e62bc2645526496a650c0e05899aa4a#diff-2dd5d7be7091937effd377c66577e7fb

**Comment by dan-online** [comment]
I saw in the file but you didn't add a test for it that I can see working in circleci

File: src/test/test.js [hunk 1]
```diff
@@ -19,7 +19,13 @@ describe("Spring.js", () => {
...
       viewsDir: `${__dirname}/views`,
-      publicDir: `${__dirname}/public`
+      publicDir: `${__dirname}/public`,
+      routes: [
+        {
+          url:"/api",
+          router: require('./routes/api')
+        }
+      ]
     });
```
Comment (dan-online) [inline src/test/test.js:29]: @DivySrivastava needs to have an it function below too for testing like the get request testing

**Comment by littledivy** [comment]
@dan-online
Oh is it something related to circleCI?
I don't have much experience in it. Can you add the test for it?
Thanks

**Comment by dan-online** [comment]
Ok I've added it myself, look at [a0fe7be](https://github.com/dan-online/spring.js/pull/1/commits/a0fe7be1f41c85d5ff6238ac5051e58d41af5f6c) for future reference

File: src/main/server.js [hunk 2]
```diff
@@ -33,7 +33,11 @@ module.exports = function(options) {
   app.use(cookieParser());
   app.use(express.static(options.publicDir));
   // App.use(require("../index").app);
-
+if(options.routes) {
```
Comment (dan-online) [inline src/main/server.js]: please add a check for array and checks for if the files have module exports and that it is a router, use throw TypeError("descriptive message")

**Review by dan-online — CHANGES_REQUESTED** [review]

File: src/main/server.js (hunk 2 above)
Comment (littledivy) [reply src/main/server.js]: Yes

File: src/test/test.js (hunk 1 above)
Comment (littledivy) [reply src/test/test.js:29]: Ohhhhh right

**Comment by littledivy** [comment]
@dan-online I've added the array check and router validity.

**Review by dan-online — APPROVED** [review]
I will merge and publish once you add this function to the docs branch in a different pull request

**Comment by littledivy** [comment]
Ok will do that

**Comment by littledivy** [comment]
Added docs...pls see #3

**Review by dan-online — CHANGES_REQUESTED** [review]
Please add an ajax request to make sure the routes are working

**Comment by dan-online** [comment]
Also, Codacy is not up to standards, please fix that

**Comment by littledivy** [comment]
@dan-online should be good to go now...

**Pull request merged**